pkg/pub_integration/lib/script/publisher.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) 2019, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+
+import '../src/pub_http_client.dart';
+import '../src/pub_tool_client.dart';
+import '../src/test_data.dart';
+
+typedef InviteCompleterFn = Future Function();
+
+/// A single object to execute publisher-related integration script and
+/// verification tests with the `pub` tool on the fake site.
+class PublisherScript {
+  final String pubHostedUrl;
+  final String credentialsFileContent;
+  PubHttpClient _pubHttpClient;
+  PubToolClient _pubToolClient;
+
+  Directory _temp;
+
+  PublisherScript({
+    @required this.pubHostedUrl,
+    @required this.credentialsFileContent,
+  });
+
+  /// Verify all integration steps.
+  Future verify() async {
+    assert(_pubHttpClient == null);
+    assert(_pubToolClient == null);
+    _pubHttpClient = PubHttpClient(pubHostedUrl);
+    _temp = await Directory.systemTemp.createTemp('pub-integration');
+    try {
+      _pubToolClient = await PubToolClient.create(
+          pubHostedUrl: pubHostedUrl,
+          credentialsFileContent: credentialsFileContent);
+
+      await _publishVersion('1.0.0');
+      await Future.delayed(Duration(seconds: 1));
```

Comment:
If this is waiting for the server to be ready.. caches to be cleared, etc... why not do polling instead.

So instead of sleep and calling `_verifyDummyPkg`, we call `await _verifyDummyPkg(version: '1.0.0');` until it successful..
example:
```dart
await poll(
  () => _verifyDummyPkg(version: '1.0.0'),
  timeout: Duration(seconds: 10),
);
```
then in `utils.dart` or somewhere else we define:
```dart
Future<T> poll<T>(T Function() fn, {@required Duration timeout, Duration interval = const Duration(seconds: 2)}) async {
  // something about a timer..
  while(timer.elapsed > timeout) {
    try {
      return await fn();
    } on Exception {
      await Future.delayed(interval);
    }
  }
}
```

or something like that... The idea being that this is more robust...

Reply:
It is not required at the moment, but I had plans to also include search in the integration tests, and that would take some time to happen. I think polling makes this a bit more complicated than needed, if it doesn't happen in 1 second, it should be considered not happening :)